src/components/context-menu.tsx

```diff
@@ -0,0 +1,173 @@
+
+import { control } from '../redux/actions';
+import { actions } from '../redux/context-menu-feature';
+
+import { useShallowEqualSelector } from '../frontend-utils';
+import { makeStyles } from 'tss-react/mui';
+import { Box, Button, ButtonProps } from '@mui/material';
+import { Capability } from '../services/interfaces/netmd';
+import { Delete, Edit, PlayArrow } from '@mui/icons-material';
+
+interface ContextMenuProps {
+    onTogglePlayPause: (event: React.MouseEvent, trackIdx: number) => void;
+    onRename: (event: React.MouseEvent, trackIdx: number) => void;
+    onDelete: (event: React.MouseEvent, trackIdx: number) => void;
+}
+
+interface ContextButtonProps extends ButtonProps {
+    icon?: React.ReactNode;
+}
+
+const useStyles = makeStyles()((theme) => ({
+    background: {
+        position: 'fixed',
+        top: 0,
+        left: 0,
+        width: '100%',
+        height: '100%',
+        backgroundColor: 'rgba(0, 0, 0, 0.3)',
+        zIndex: 998,
+        display: 'flex',
+        justifyContent: 'center',
+    },
+    reopenArea: {
+        width: '700px',
+        height: '700px',
+        zIndex: 999,
+    },
+    menuContainer: {
+        zIndex: 1000,
+        boxSizing: 'border-box',
+        position: 'fixed',
+        width: '160px',
+        backgroundColor: '#303030',
+        borderRadius: theme.shape.borderRadius,
+        color: 'white',
+        padding: '2px',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-start',
+        border: '1px solid #444',
+    },
+    button: {
+        gap: '5px',
+        color: 'white',
+        backgroundColor: 'transparent',
+        display: 'flex',
+        alignItems: 'center',
+        border: 'none',
+        cursor: 'pointer',
+        padding: '2px',
+        textAlign: 'left',
+        height: '24px',
+        fontSize: '10px',
+        width: '100%',
+        justifyContent: 'flex-start',
+        '&:hover': {
+            backgroundColor: '#444',
+        },
+    },
+}));
+
+const ContextButton = ({ icon, children, ...otherProps }: ContextButtonProps) => {
+    const {
+        classes: { button },
+    } = useStyles();
+
+    return (
+        <Button className={button} {...otherProps}>
+            {icon}
+            {children}
+        </Button>
+    );
+};
+
+export const ContextMenu = ({ onTogglePlayPause, onRename, onDelete }: ContextMenuProps) => {
+    const dispatch = useDispatch();
+
+    const { classes } = useStyles();
+
+    const position = useShallowEqualSelector((state) => state.contextMenu.position);
+    const isVisible = useShallowEqualSelector((state) => state.contextMenu.visible);
+    const contextTrack = useShallowEqualSelector((state) => state.contextMenu.track);
+
+    const deviceCapabilities = useShallowEqualSelector((state) => state.main.deviceCapabilities);
+
+    const isCapable = useCallback((capability: Capability) => deviceCapabilities.includes(capability), [deviceCapabilities]);
```

Comment:
it occurs to me that we might be able to pull these lines (`isCapable`, `handlePlayTrack` etc) out and make a hook that can be reused, since it's the same code in a couple of different places.

LMK!